src/web/api_handlers.c

```diff
@@ -4,57 +4,176 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <ctype.h>
 
 #include "web/api_handlers.h"
-#include "web/web_server.h"
+#include "web/mongoose_adapter.h"
+#include "web/http_router.h"
+#include "web/mongoose_server_handlers.h"
 #include "core/logger.h"
-#include "web/api_handlers_detection_results.h"
+#include "core/config.h"
+#include "mongoose.h"
+
+// We'll use the existing mg_handle_* functions directly
+
+/**
+ * @brief Register direct Mongoose API handlers
+ */
+void register_api_handlers(struct mg_mgr *mgr) {
+    if (!mgr) {
+        log_error("Invalid Mongoose manager for registering direct API handlers");
+        return;
+    }
+    
+    log_info("Registering direct Mongoose API handlers");
+    
+    // Note: The actual registration happens in mongoose_server.c in the s_api_routes table
+    // This function is called during server initialization to perform any additional setup
+    
+    // Log the handlers that are being registered
+    log_info("Registered handler: GET /api/streams");
```

Comment:
Why not use a data structure? This will quickly not be maintainable.

Reply:
I truthfully think this logging should be removed.